File: Lab/Lab_1_2/src/Lab_1/ReadFileTest.py
import unittest

from src.Lab_1.ReadFile import ReadFile


class TestReadFile(unittest.TestCase):
    EMPTY_FILE = "/home/leym1ion/IdeaProjects/Lab_1_Python/src/main/resources/EmptyTest.txt"
    NON_EMPTY_FILE = "/home/leym1ion/IdeaProjects/Lab_1_Python/src/main/resources/TextTest.txt"
    ZIP_FILE = "/home/leym1ion/IdeaProjects/Lab_1_Python/src/main/resources/Test.zip"

    def test_is_file_empty_should_throw_exception_for_zip_file(self):
        with self.assertRaises(ValueError) as context:
            ReadFile.is_file_empty(self.ZIP_FILE)
        self.assertIn("Only txt File", str(context.exception))

    def test_file_exists_should_return_false_for_missing_file(self):
        self.assertFalse(ReadFile.file_exists("Baihgui_File.txt"))

    def test_is_file_empty_should_return_true_for_empty_test_file(self):
        self.assertTrue(ReadFile.is_file_empty(self.EMPTY_FILE))

    def test_is_file_empty_should_return_false_for_non_empty_test_file(self):
        self.assertFalse(ReadFile.is_file_empty(self.NON_EMPTY_FILE))


if __name__ == "__main__":
    unittest.main()

File: Lab/Lab_1_1/src/test/java/Lab_1/ReadFileTest.java
package Lab_1;

import ip.group.Lab_1.ReadFile;
import org.junit.jupiter.api.Test;

import java.io.File;

import static org.junit.jupiter.api.Assertions.*;

public class ReadFileTest {

    @Test
    public void splitNumberTest() {
        String str = "1 2 3 4 5 6";

        int[] resultArr = ReadFile.toNumberSplit(str);
        assertArrayEquals(new int[]{1, 2, 3, 4, 5, 6}, resultArr);
    }

    @Test
    public void splitNumberNewLineSpace() {
        String str = "1 \n 2 \n 3   4 5 6";
        int[] resultArr = ReadFile.toNumberSplit(str);
        assertArrayEquals(new int[]{1, 2, 3, 4, 5, 6}, resultArr);
    }

    @Test
    public void splitNumberInvalidFormat() {
        String str = "1 2 3 a 4 5 6";

        assertThrows(NumberFormatException.class, () -> {
            ReadFile.toNumberSplit(str);
        });
    }


    File emptyFile = new File("src/main/resources/EmptyTest.txt");
    File nonEmptyFile = new File("src/main/resources/TextTest.txt");
    File zipFile = new File("src/main/resources/Test.zip");

    @Test
    void isFileEmptyShouldThrowExceptionForZipFile() {
        assertThrows(IllegalArgumentException.class, () -> {
            ReadFile.isFileEmpty(zipFile.getAbsolutePath());
        });
    }

    @Test
    void fileExistsShouldReturnFalseForMissingFile() {
        assertFalse(ReadFile.fileExists("Baihgui_File.txt"));
    }

    @Test
    void isFileEmptyShouldReturnTrueForEmptyTestFile() {
        assertTrue(ReadFile.isFileEmpty(emptyFile.getAbsolutePath()));
    }

    @Test
    void isFileEmptyShouldReturnFalseForNonEmptyTestFile() {
        assertFalse(ReadFile.isFileEmpty(nonEmptyFile.getAbsolutePath()));
    }
}
